### PR TITLE
Add parameter verification for api creation network

### PR DESCRIPTION
**Pull request by zhangguanzhang**
Signed-off-by: zhangguanzhang <zhangguanzhang@qq.com>
In the absence of parameter verification, an incorrect network will be created https://github.com/containers/podman/issues/7262#issuecomment-670967217

### CODE REVIEW

**Comment by zhangguanzhang** [comment]
I test this is ok on my local machine.......
```
[root@sz-test apiv2]# ./test-apiv2 35
ok 1 [35-networks] GET networks/non-existing-network : status=404
ok 2 [35-networks] POST libpod/networks/create [] : status=201
ok 3 [35-networks] POST libpod/networks/create ["Subnet":{"IP":"10.10.254.0","Mask":[255,255,255,0]}] : status=201
ok 4 [35-networks] POST libpod/networks/create ["Subnet":{"IP":"10.10.1.0","Mask":[]}] : status=500
ok 5 [35-networks] POST libpod/networks/create ["Subnet":{"IP":"10.10.1.0","Mask":[0,255,255,0]}] : status=500
1..5
[root@sz-test apiv2]# 
```

**Comment by danopia** [comment]
A couple of the failing test runs are rootless... which can't use CNI networks. You might need to figure out how to isolate the network creation tests to rootfull runs.

**Comment by rhatdan** [comment]
SkipIfRootless()

**Comment by zhangguanzhang** [comment]
@danopia @rhatdan thanks

**Comment by rhatdan** [comment]
@baude @edsantiago @jwhonce @mheon PTAL

File: pkg/api/server/register_networks.go [hunk 1]
```diff
@@ -179,7 +179,7 @@ func (s *APIServer) registerNetworkHandlers(r *mux.Router) error {
 	//    schema:
 	//      $ref: "#/definitions/NetworkCreateOptions"
 	// responses:
-	//   200:
+	//   201:
```
Comment (TomSweeneyRedHat) [inline pkg/api/server/register_networks.go:182]: are we consistent with 200/201 across all successful create responses?  Have we already set expectations for 200 with end users?  I prefer the 201, but don't want to trip folks up.

File: test/apiv2/35-networks.at [hunk 2]
```diff
@@ -3,6 +3,26 @@
 # network-related tests
 #
 
-t GET /networks/non-existing-network 404
+t GET networks/non-existing-network 404
+
+if root; then
+  t POST libpod/networks/create?name=network1 '' 200 \
+    .Filename~.*\\.conflist
```
Comment (edsantiago) [inline test/apiv2/35-networks.at:10]: Nit (but not worth re-pushing): now that you're creating a named network, you can check for `/network1.conflist`. Same below. Please don't re-push just for this, though.

**Comment by zhangguanzhang** [comment]
all had done

**Pull request merged**